php/class-plugin.php

```diff
@@ -21,19 +21,45 @@ class Plugin extends Plugin_Abstract {
 	public $admin;
 
 	/**
-	 * Execute this once plugins are loaded. (not the best place for all hooks)
+	 * Block Post Type.
+	 *
+	 * @var Post_Types\Block_Post
 	 */
-	public function plugin_loaded() {
-		$this->admin = new Admin\Admin();
-		$this->register_component( $this->admin );
-	}
+	public $block_post;
 
 	/**
-	 * Check if a valid Pro license has been activated on this site.
+	 * Initiate the loading of new blocks.
 	 *
-	 * @return bool
+	 * @var Blocks\Loader
+	 */
+	public $loader;
+
+	/**
+	 * Execute this as early as possible.
+	 */
+	public function init() {
+		$this->set_util();
```

Comment:
@kienstra – how do you feel about implementing this here? Is it better to do it in `block-lab.php` instead? What setting the `Util` in a parent `init` method?

Reply:
Hi @lukecarbis,

To go along with this `init()` method, maybe it'd be good to simply remove `set_util()` and put the function's contents here:

```php
$this->util = new Util();
$this->register_component( $this->util );
```

Reply:
PR #384, targeting this PR's branch, implements this. It also changes the way the post type slug is stored. 

This is your PR, so it's up to you whether to use that approach, or part of it. I didn't want to push straight to this branch.